codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java

```diff
@@ -255,6 +261,42 @@ public void entityRequestEntityResponse() throws Exception {
         assertTrue(channel.isActive());
     }
 
+    @Test
+    public void pushPromiseSend() throws Exception {
+        frameListener.onHeadersRead(http2HandlerCtx, 1, request, 0, false);
+        Http2HeadersFrame inboundHeaders = inboundHandler.readInbound();
+        Http2FrameStream stream = inboundHeaders.stream();
+
+        Http2PushPromiseFrame frame = new DefaultHttp2PushPromiseFrame(request).stream(stream);
+        inboundHandler.writeOutbound(frame);
+
+        verify(frameWriter).writePushPromise(eq(http2HandlerCtx), eq(stream.id()), eq(2),
+                any(Http2Headers.class), anyInt(), anyChannelPromise());
+
+        assertEquals(2, frame.promisedStream().id());
+    }
+
+    @Test
+    public void pushPromiseReceive() throws Exception {
+        final int PUSH_PROMISE_STREAM_ID = 2;
+
+        Http2Settings settings = new Http2Settings();
+        setUp(Http2FrameCodecBuilder.forClient(), settings);
+        Http2FrameStream frameStream = frameCodec.newStream();
+
+        inboundHandler.writeOutbound(new DefaultHttp2HeadersFrame(request, true).stream(frameStream));
+        frameListener.onPushPromiseRead(http2HandlerCtx, frameStream.id(), PUSH_PROMISE_STREAM_ID, request, 0);
+        Http2PushPromiseFrame frame = inboundHandler.readInbound();
+        assertNotNull(frame);
+        assertEquals(PUSH_PROMISE_STREAM_ID, frame.promisedStream().id());
+        assertEquals(frameStream.id(), frame.stream().id());
```

Comment:
Would it be better to assert that the streams themselves were equal, not just their id's?